web/modules/custom/bnf/src/Plugin/FieldTypeTraits/EmbedVideoTrait.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Drupal\bnf\Plugin\FieldTypeTraits;
+
+use Drupal\autowire_plugin_trait\AutowirePluginTrait;
+use Drupal\bnf\GraphQL\Operations\GetNode\Node\Paragraphs\EmbedVideo\MediaVideo;
+use Drupal\bnf\GraphQL\Operations\GetNode\Node\Paragraphs\EmbedVideo\MediaVideotool;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Spawnia\Sailor\ObjectLike;
+
+/**
+ * Helper trait, for dealing with Embedded video fields.
+ */
+trait EmbedVideoTrait {
+  use AutowirePluginTrait;
+
+  public function __construct(
+    protected EntityTypeManagerInterface $entityTypeManager,
+  ) {
+  }
+
+  /**
+   * Getting Drupal-ready value from object.
+   *
+   * @return mixed[]
+   *   The value that can be used with Drupal ->set().
+   */
+  public function getEmbedVideoValue(MediaVideo|MediaVideotool|ObjectLike|null $video): array {
+    if (is_null($video)) {
+      return [];
+    }
+
+    $mediaStorage = $this->entityTypeManager->getStorage('media');
+
+    if (str_ends_with(get_class($video), 'MediaVideo')) {
+      /** @var \Drupal\bnf\GraphQL\Operations\GetNode\Node\Paragraphs\EmbedVideo\MediaVideo $video */
+
+      $media = $mediaStorage->create([
```

Comment:
We'll need to deal with this when supporting updates.

Reply:
We need to test it. This will create a new media entry on each update, but we need to figure out what happens when the cron job cleans out orphaned paragraphs. Does the `entity_reference` field delete the now unreferenced media entity (I somewhat doubt it)?

Alternatively, we need to check if there's already a media entity with the given values, and reuse it if that's the case. It means that if there are two media entities with the same data, they'll be turned into one, but that could be considered a feature.